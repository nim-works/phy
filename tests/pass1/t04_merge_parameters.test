discard """
  description: "Test with split control-flow paths"
"""
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) 0
    (List
      (Block (Params) (Locals)
        (Branch (IntVal 1)
          (Goto 1 (List))
          (Goto 2 (List))))

      (Block (Params)
        (Locals (Type 0))
        (Asgn (Local 0) (IntVal 100))
        (Goto 3
          (List (Local 0))))
      (Block (Params)
        (Locals (Type 0))
        (Asgn (Local 0) (IntVal 200))
        (Goto 3
          (List (Local 0))))

      (Block (Params (Type 0))
        (Locals)
        (Drop
          (Move (Param 0)))
        (Return)))))
