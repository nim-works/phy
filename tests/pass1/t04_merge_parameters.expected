;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) 0
    (Locals (Type 0))
    (List
      (Block (Params)
        (Branch (IntVal 1)
          (Goto 1)
          (Goto 2)))
      (Block (Params)
        (Asgn (Local 0) (IntVal 100))
        (Goto 3))
      (Block (Params)
        (Asgn (Local 0) (IntVal 200))
        (Goto 3))
      (Block (Params)
        (Drop
          (Move (Local 0)))
        (Return)))))